### PR TITLE
add rccl to tf.contrib.distribute

### PR DESCRIPTION
**Pull request by jeffdaily**

### CODE REVIEW

File: tensorflow/contrib/distribute/python/cross_tower_utils.py [hunk 1]
```diff
@@ -21,7 +21,11 @@
 import collections as pycoll
 import threading
 
-from tensorflow.contrib import nccl
+from tensorflow.python import pywrap_tensorflow
+if pywrap_tensorflow.IsGoogleCudaEnabled():
+  from tensorflow.contrib import nccl
+else:
+  from tensorflow.contrib import rccl as nccl
```
Comment (whchung) [inline tensorflow/contrib/distribute/python/cross_tower_utils.py]: should this be checked with `IsBuiltWithROCm`?

Comment (whchung) [reply tensorflow/contrib/distribute/python/cross_tower_utils.py]: please refer to `tensorflow/core/util/port.h`

Comment (jeffdaily) [reply tensorflow/contrib/distribute/python/cross_tower_utils.py]: This is the exact same code stub from tf.contrib.all_reduce, so if this needs to change it should be updated there, as well.  I recall asking how to do this check on slack but I can't find the discussion now.

Comment (whchung) [reply tensorflow/contrib/distribute/python/cross_tower_utils.py]: @jeffdaily yeah I think it's worthwhile to change both places for the logic be upstreamed. please also check `tensorflow/python/framework/test_util.py` you can see such python API has been added.

Comment (jeffdaily) [reply tensorflow/contrib/distribute/python/cross_tower_utils.py]: Would it be appropriate to piggyback on this PR to update both places?

Comment (whchung) [reply tensorflow/contrib/distribute/python/cross_tower_utils.py]: sure thing. please do so.

**Pull request merged**